trackintel/preprocessing/positionfixes.py

```diff
@@ -386,3 +336,91 @@ def generate_triplegs(positionfixes, staypoints=None, *args, **kwargs):
+                                pfs[j][name_geocol].x, pfs[j][name_geocol].y)
+
+            if dist > dist_threshold:
+                delta_t = pfs[j]['tracked_at'] - pfs[i]['tracked_at']
+                if delta_t.total_seconds() > time_threshold:
+                    staypoint = {}
+                    staypoint['user_id'] = pfs[i]['user_id']
+                    staypoint[name_geocol] = Point(np.mean([pfs[k][name_geocol].x for k in range(i, j)]),
+                                                    np.mean([pfs[k][name_geocol].y for k in range(i, j)]))
+                    if elevation_flag:
+                        staypoint['elevation'] = np.mean([pfs[k]['elevation'] for k in range(i, j)])
+                    staypoint['started_at'] = pfs[i]['tracked_at']
+                    staypoint['finished_at'] = pfs[j - 1][
+                        'tracked_at']  # TODO: should this not be j-1? because j is not part of the staypoint. DB: Changed.
+
+                    # store matching, index should be the id of pfs
+                    staypoint['pfs_id'] = [pfs[k]['id'] for k in range(i, j)]
+
+                    # add staypoint
+                    ret_spts = ret_spts.append(staypoint, ignore_index=True)
+
+                    # TODO Discussion: Is this last point really a staypoint? As we don't know if the
+                    #      person "moves on" afterwards...
+                    if j == num_pfs - 1:
+                        staypoint = {}
+                        staypoint['user_id'] = pfs[j]['user_id']
+                        staypoint[name_geocol] = Point(pfs[j][name_geocol].x, pfs[j][name_geocol].y)
+                        if elevation_flag:
+                            staypoint['elevation'] = pfs[j]['elevation']
+                        staypoint['started_at'] = pfs[j]['tracked_at']
+                        staypoint['finished_at'] = pfs[j]['tracked_at']
+                        # store matching, index should be the id of pfs
+                        staypoint['pfs_id'] = [pfs[j]['id']]
+
+                        ret_spts = ret_spts.append(staypoint, ignore_index=True)
+                i = j
+                break
+            j = j + 1
+    
+    return ret_spts
+
+def _generate_staypoints_dbscan_user(df, 
+                                     name_geocol, 
+                                     epsilon = 100, 
+                                     num_samples = 1):
+    db = DBSCAN(eps=epsilon / 6371000, min_samples=num_samples, algorithm='ball_tree', metric='haversine')
+
+
+    # TODO: enable transformations to temporary (metric) system
+    transform_crs = None
+    if transform_crs is not None:
+        pass
+
+    # get staypoint matching
+    p = np.array([[radians(g.y), radians(g.x)] for g in df[name_geocol]])
+    labels = db.fit_predict(p)
+
+    # add positionfixes - staypoint matching to original positionfixes
+    df['staypoint_id'] = labels
```

Comment:
maybe you could rename "df" to "pfs" as you did in the sliding function. I think this might be easier to read (if the function is only used for pfs...)